### PR TITLE
upstream stuff to code_coverage

### PR DESCRIPTION
**Pull request by mikeferguson**
Also no longer need to modify source

### CODE REVIEW

File: example_ws/src/example_pkg/package.xml [hunk 1]
```diff
@@ -60,7 +60,7 @@
   <exec_depend>rospy</exec_depend>
   <exec_depend>std_msgs</exec_depend>
 
-  <test_depend>python-coverage</test_depend>
+  <test_depend>code_coverage</test_depend>
```
Comment (SarvagyaVaish) [inline example_ws/src/example_pkg/package.xml:63]: Is this dependency met by the clone of the code_coverage repo, which is specified via the upstream.repos file?

File: example_ws/src/example_pkg/test/sample_rostest.test [hunk 2]
```diff
@@ -1,10 +1,14 @@
 <launch>
 
-    <env name="COVERAGE_PROCESS_START" value=".coveragerc"/>
+    <arg name="coverage" default="false"/>
```
Comment (SarvagyaVaish) [inline example_ws/src/example_pkg/test/sample_rostest.test:3]: During tests, we want this to default to true, right?
We don't necessarily need to upload the results, but my workflow (at least in a past job that cared about coverage) was to generate the coverage report, examine it for gaps and then write more tests.

File: .travis.yml [hunk 3]
```diff
@@ -15,9 +15,12 @@ jobs:
...
         - ADDITIONAL_DEBS="curl"
         - ROS_DISTRO="melodic"
-        - BEFORE_SETUP_TARGET_WORKSPACE="pip install coverage"
-        - DOCKER_RUN_OPTS="-e TRAVIS_COMMIT=$TRAVIS_COMMIT -e COVERAGE_RCFILE=/root/target_ws/src/ros_ci_and_coverage/.coveragerc -e COVERAGE_PROCESS_START=.coveragerc -e CODECOV_TOKEN=$CODECOV_TOKEN"
-        - AFTER_RUN_TARGET_TEST="cd /root/.ros && coverage combine && coverage xml"
+        - UPSTREAM_WORKSPACE="upstream.repos"
+        - CMAKE_ARGS="-DENABLE_COVERAGE_TESTING=ON -DCMAKE_BUILD_TYPE=Debug"
+        - DOCKER_RUN_OPTS="-e TRAVIS_COMMIT=$TRAVIS_COMMIT -e CODECOV_TOKEN=$CODECOV_TOKEN"
+        # Manually trigger combine (we aren't actually calling code_coverage target)
+        - AFTER_RUN_TARGET_TEST="cd /root/.ros && python-coverage combine && python-coverage xml"
```
Comment (SarvagyaVaish) [inline .travis.yml:22]: Oh.. if we're not calling the code_coverage target, what does the dependency do?

File: example_ws/src/example_pkg/package.xml (hunk 1 above)
Comment (mikeferguson) [reply example_ws/src/example_pkg/package.xml:63]: yes it is (I added python-coverage there, we don't actually directly call python-coverage at all from this repo)

File: example_ws/src/example_pkg/test/sample_rostest.test (hunk 2 above)
Comment (mikeferguson) [reply example_ws/src/example_pkg/test/sample_rostest.test:3]: if ENABLE_COVERAGE_TESTING is set, this gets set true. That'd be the "standard" way of doing it (so it isn't always generating coverage). However, you could certainly set this to tru if you wanted - but note that it gets overwritten by the cmake (setting it equal to ENABLE_COVERAGE_TESTING)

File: .travis.yml (hunk 3 above)
Comment (mikeferguson) [reply .travis.yml:22]: it still does the setup for nosetests - and also does the C++ stuff (not relevant to this repo - but relevant to stuff down the road)

Comment (mikeferguson) [reply .travis.yml:22]: Also - we can use that target to call individual packages and set thresholds for them (something I've not done here yet) in an AFTER_SCRIPT

**Review by SarvagyaVaish — APPROVED** [review]

**Pull request merged**